wrapanapi/systems/redfish.py

```diff
@@ -15,16 +17,30 @@ class RedfishSystem(System):
         password: The password to connect with.
     """
 
+    # statistics for the provider
     _stats_available = {
-        'num_server': lambda self: 1,
+        'num_server': lambda self: self.num_servers(),
+    }
+
+    # statistics for an individual server
+    _server_stats_available = {
+        'cores_capacity': lambda self, requester: self.get_server_cores(requester.name),
+        'memory_capacity': lambda self, requester: self.get_server_memory(requester.name),
+        'num_firmwares': lambda self, requester: self.get_num_server_firmwares(requester.name),
+        'num_network_devices': lambda self, requester: self.get_num_network_devices(requester.name),
+        'num_storage_devices': lambda self, requester: self.get_num_storage_devices(requester.name),
     }
 
-    def __init__(self, hostname, username, password, protocol="https", api_port=443, **kwargs):
-        super(RedfishSystem, self).__init__(kwargs)
-        self.api_port = api_port
-        self.auth = (username, password)
-        self.url = '{}://{}:{}/'.format(protocol, hostname, self.api_port)
+    _server_inventory_available = {
+        'power_state': lambda self, requester: self.get_server_power_status(requester.name),
+    }
+
+    def __init__(self, hostname, username, password, security_protocol, api_port=443, **kwargs):
```

Comment:
`security_protocol` became required parameter and this means that doctring for this class should be updated accordingly.